addons/auth_totp/views/user_preferences.xml

```diff
@@ -1,51 +1,64 @@
 <odoo>
-    <record model="ir.ui.view" id="view_totp_list">
-        <field name="name">users list: add totp status</field>
+    <record id="res_users_view_search" model="ir.ui.view">
+        <field name="name">res.users.view.search.inherit.auth.totp</field>
         <field name="model">res.users</field>
-        <field name="inherit_id" ref="base.view_users_tree"/>
+        <field name="inherit_id" ref="base.view_users_search" />
         <field name="arch" type="xml">
-            <tree>
-                <field name="totp_enabled"/>
-            </tree>
+            <search string="Users">
+                <separator/>
+                <filter name="totp_enabled" string="2-Step authentication Enabled" domain="[('totp_secret','!=',False)]"/>
+                <separator/>
+                <filter name="totp_disabled" string="2-Step authentication Disabled" domain="[('totp_secret','=',False)]"/>
+            </search>
         </field>
     </record>
+
     <record model="ir.ui.view" id="view_totp_form">
         <field name="name">user form: add totp status</field>
         <field name="model">res.users</field>
         <field name="inherit_id" ref="base.view_users_form"/>
         <field name="arch" type="xml">
-            <xpath expr="//page[@name='references']/group[1]" position="before">
-                <field name="totp_enabled" invisible="1"/>
-                <group attrs="{'invisible': [('totp_enabled', '!=', False)]}">
-                    <div>
-                        <span class="alert alert-info" role="status">
-                            <i class="fa fa-warning"/>
-                            Two-factor authentication not enabled
-                        </span>
-                    </div>
-                </group>
-                <group attrs="{'invisible': [('totp_enabled', '=', False)]}">
-                    <div>
-                        <span class="text-success">
-                            <i class="fa fa-check-circle"/>
-                            Two-factor authentication enabled
+            <xpath expr="//page[@name='preferences']" position="after">
+                <page string="Account Security" name="security">
+                    <field name="totp_enabled" invisible="1"/>
+                    <!-- For own user, allow to activate the 2-Step Authentication -->
+                    <group>
+                        <div>
+                            <div class="o_horizontal_separator d-flex align-items-center mt-0">2-Step Authentication
+                                <div attrs="{'invisible': [('totp_enabled', '!=', False)]}">
+                                    <button attrs="{'invisible': &quot;[('id', '=', uid)]&quot;}" name="totp_enable_wizard"
+                                            disabled="1" type="object" class="fa fa-toggle-on o_auth_2fa_btn disabled" aria-label="Enable TOTP"></button>
```

Comment:
I hereby stand that effectively I would avoid custom classes and css in form views... maybe, if reaaaaaally impossible to avoid, standard bootstrap classes (primary, light, ...).

Reply:
this won't work as 'custom style' is not transposable to Odooish class or standard BS classes. unless i'm missing something. Once again, custom button or not at all ? @LNAodoo